### PR TITLE
Pruning Integration

### PR DESCRIPTION
**Pull request by rohanali1798**
Added dummy script file in project directory in first commit.

@kinow

### CODE REVIEW

**Comment by kinow** [comment]
We have a single pull request now, @rohanali1798 , @yolandab , @suvarchal .

I will push my changes here and test on MN4, then merge it once it works! :tada:

**Comment by kinow** [comment]
Default is `1` member.

```bash
(venv) (base) bdepaula@bsces107921:~/Development/python/workspace/workflow-registry/Pillar_II/esm/src$ ./launch_fesom2.sh --hpc mn4 --start_dates 1948 -c fesom2/esm_ensemble.conf --cores_per_node=48 --cores=144
Launching fesom2 eFlows4HPC ESM experiment...🚀

Launch arguments:

EXPID           : d18308
MODEL           : fesom2
HPC             : mn4
CORES           : 144
CORES PER NODE  : 48
NODES           : 3
QOS             : 
START DATES     : (1) 1948
MEMBERS         : 1
/home/bdepaula/Development/python/workspace/workflow-registry/Pillar_II/esm/src/fesom2/env/mn4.sh: line 7: module: command not found
```

To specify more members, the argument will be `--members` (@rohanali1798 you can start updating the topology if you'd like):

```bash
(venv) (base) bdepaula@bsces107921:~/Development/python/workspace/workflow-registry/Pillar_II/esm/src$ ./launch_fesom2.sh --hpc mn4 --start_dates 1948 -c fesom2/esm_ensemble.conf --cores_per_node 48 --cores 144 --members 2
Launching fesom2 eFlows4HPC ESM experiment...🚀

Launch arguments:

EXPID           : f10929
MODEL           : fesom2
HPC             : mn4
CORES           : 144
CORES PER NODE  : 48
NODES           : 3
QOS             : 
START DATES     : (1) 1948
MEMBERS         : 2
/home/bdepaula/Development/python/workspace/workflow-registry/Pillar_II/esm/src/fesom2/env/mn4.sh: line 7: module: command not found
```

Working on the Python code now, to verify the name lists are updated.

**Comment by kinow** [comment]
Ensemble member generation, ocean data perturbation, and running one `fesom.x` per member implemented. Let's wait now for CI...

File: Pillar_II/esm/src/fesom2/__init__.py [hunk 1]
```diff
@@ -68,13 +80,25 @@ def _namelists(start_date: str, config: ConfigParser):
     Returns:
         A dictionary with the name of the namelist, and an optional dictionary of values to process the namelist.
     """
-    model_config = _get_model_config(start_date, config)
+    model_config = _get_model_config(start_date, member, config)
+    temp_min = random.uniform(-1.0, 2.0)
+    temp_max = random.uniform(2.0, 4.0)
+    # N.B.: Salinity must be above 1.
+    sal_min = random.uniform(1.5, 2.5)
+    sal_max = random.uniform(3.0, 4.0)
```
Comment (rohanali1798) [inline Pillar_II/esm/src/fesom2/__init__.py:88]: Hi @suvarchal 

We tested this and it caused a model blow up.
Screenshot from 2024-02-22 17-15-47

Link to paper https://gmd.copernicus.org/articles/7/663/2014/gmd-7-663-2014.pdf
[5:17](https://bsc-es.slack.com/archives/D059GDB8HJ6/p1708618662225279)
When the local sea surface salinity is far from
the reference salinity, virtual salt flux formulation can lead to
too small or too large salinity and thus model blowup.

We think this is the problem. However, another run worked but maybe the values were in the range luckily?

**Comment by kinow** [comment]
I'm suspecting the current issue with the pruning task is around here:

```py
        mdc = MetaDictClass.get_by_alias(expid)
        if mdc:
            logging.info("Breaking while-loop")
            break
```

I think `get_by_alias` never returns a Truthy value. Couldn't find docs/tests by looking at the code used in the IDE, so asked @yolandab via email.

If that's really the case, it might be easy to fix and get pruning working! :tada: :crossed_fingers:

**Review by kinow — APPROVED** [review]
Both @rohanali1798 and I tested and the pruning is working on MN4.

**Pull request merged**